addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsPipelineRegistry.java

```diff
@@ -13,32 +33,91 @@
  */
 @ApplicationScoped
 public class JenkinsPipelineRegistry {
+    private Map<String, JenkinsPipeline> pipelines = Collections.emptyMap();
```

Comment:
Why store in a Map and not in a Collection?

Reply:
If the intention is to prevent duplicate IDs, just implement in the interface directly: http://immutables.github.io/immutable.html#custom-equals-hashcode-tostring

Reply:
Better, make the getId method as: 
`@Value.Derived

Reply:
Ah but we need an Id to pass on to the projectile, nevermind, the ID is necessary

Reply:
For the `getById()`

Reply:
Could have done a stream filter of course, but that's really inefficient, so I just preferred doing it this way.

Reply:
Nah, that's fine. Much faster